seed/seed20.go

```diff
@@ -1113,14 +1154,31 @@ func (s *seed20) lookupSnap(snapRef naming.SnapRef, modelSnap *asserts.ModelSnap
 	} else {
 		var err error
 		timings.Run(tm, "derive-side-info", fmt.Sprintf("hash and derive side info for snap %q", snapRef.SnapName()), func(nested timings.Measurer) {
-			path, sideInfo, seedComps, err = s.deriveSideInfo(
+			path, sideInfo, seedComps, snapRev, err = s.deriveSideInfo(
 				snapRef, modelSnap, optSnap, handler, snapsDir, tm)
 		})
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	var integrityData *integrity.IntegrityDataParams
+	var err error
+
+	if snapRev != nil {
+		timings.Run(tm, "find-integrity-params", fmt.Sprintf("find integrity params for snap %q", snapRef.SnapName()), func(nested timings.Measurer) {
```

Comment:
I'm not sure if we need to gather timing for this call, it does not look like we are deriving the Merkle tree here? Or doing any expensive operation.